microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/ContainerInstanceHolder.java

```diff
@@ -38,7 +39,12 @@ static void set(HelidonContainer container) {
         CONTAINER.set(container);
     }
 
-    static HelidonContainer get() {
+    // return true if the container was reset, indicating somebody started CDI by hand and then shut it down
+    static synchronized boolean isReset() {
+        return isReset;
+    }
```

Comment:
Is synchronized needed here?

Reply:
Yes, as the field is not atomic. Either you must use `synchronized` from all places that use the field, have it declared `volatile` or use `AtomicBoolean`. As I already used `synchronized` in the other methods and this class is used once during startup and once during shutdown, the overhead is negligible